.github/workflows/build.yml

```diff
@@ -0,0 +1,81 @@
+name: build
+on:
+  push:
+#    branches: [ master ] Todo: Uncomment before merging
+  pull_request:
+    branches: [ master ]
+  workflow_dispatch:
+    inputs:
+      release:
+        description: 'Release? yes/no'
+        default: 'no'
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+    permissions:
+      id-token: write
+      contents: read
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Configure AWS credentials
```

Comment:
For AWS auth I used the official [AWS action](https://github.com/aws-actions/configure-aws-credentials). With this approach, we don't have to configure additional identity providers in the code.